exec/vector/src/main/codegen/templates/FixedValueVectors.java

```diff
@@ -509,15 +509,15 @@ public long getTwoAsLong(int index) {
     public ${friendlyType} getObject(int index) {
       org.joda.time.DateTime date = new org.joda.time.DateTime(get(index), org.joda.time.DateTimeZone.UTC);
       date = date.withZoneRetainFields(org.joda.time.DateTimeZone.getDefault());
-      return date;
+      return new java.sql.Date(date.getMillis());
```

Comment:
Good point.  Someone with deeper knowledge should take a look.  As far as I can tell, I think the problem with SqlAccessor is that it uses the vector type to know whether to invoke getDate() vs getTimestamp().  However, such vector type knowledge is not there when complex type such as List and Map are materialized in memory to form Java JsonStringArrayList and JsonStringHashMap.  

In https://issues.apache.org/jira/browse/DRILL-6242, the example describes this scenario:

`select t.context.`date`, t.context from test t;`

where the `date` field is inside a Map type.  And we select the field by itself as well as select the Map on the same query.  This query returns:

`+--------+---------+ `
`| EXPR$0 | context | `
`+--------+---------+ `
`| 2018-03-13 | {"date":{"dayOfYear":72,"year":2018, ... |`

One can see that the first column shows the date in the right type.  But the same date is shown as a different type inside the Map.  In the vector package, when reading the [List|Map]Vector, the code produces its nested member values via the generic method "getObject()".  Since all three vector type returned the same DataObject type as the representation, there are no distinction.

For the type information to be carried within the List | Map, it would seem that the value should be of distinct types.  These can be java.sql.[Date|Time|Timestamp] or some other [Date|Time|Timestamp] classes.

Reply:
BTW, Drill internally tries to fool around with the timezone to preserve the "textual representation" look complex.  I am not convinced this is the "right" way to handle time.  But in any case, that is outside of the scope of this change.

I mentioned in the Jira a comment on how such timezone manipulation is dangerous and lead to errors.  I ran into that when attempting at creating a unit test for the change made in the pull request.

Reply:
Agree that messing with Timezones is a Bad Thing. Probably an artifact of the way java.util did things.
Anyway, I did mean using org.joda.time.Local[Data|Time|TimeStamp]  or the corresponding java.time.* classes.

Reply:
sure, I can update to the joda version or the java.time version.  Which one is preferred?  java.time.* is available in java 1.8+

Reply:
Either one is fine (since java.time is based on Joda). We've switched to Java 8, but just for consistency with the rest of the code, we might as well use Joda.

Reply:
I think updating to Java8 LocalDate/Time classes would be good choice. And it will be step forward in the resolving of the Drill's Date/Time issues mentioned in different Jiras: [DRILL-5334](https://issues.apache.org/jira/browse/DRILL-5334), [DRILL-5332](https://issues.apache.org/jira/browse/DRILL-5332) etc.

Reply:
How about we use Java 8 Local[Data|Time|Timestamp] for the public interface methods?  That sets things up for the future.

Internally, I won't change the logic that is using Joda DateTime, that is doing the various time zone stuff.  That behind the scene logic can be separately updated after determine what is the right behavior Drill wants to support.

Reply:
As I work through using Local[Date|Time|DateTime] inside the vector package, I notice that it will create the following inconsistency on the JDBC output:

SqlAccessor provides "getDate()", "getTime()", and "getTimestamp()" that are returns java.sql.[Date|Time|Timestamp].  This will convert Local[Date|Time|DateTime] into java.sql.[Date|Time|Timestamp]

For complex objects, SqlAccessor provides "getObject()" which will return JsonStringHashMap or JsonStringArrayList.  If the Local[Date|Time|DateTime] objects are inside the map and list, then they will NOT be converted into java.sql.[Date|Time|Timestamp].

Example:

`select t.context.date, t.context from test t; `

will return a java.sql.Date object for column 1, but a java.time.LocalDate for the same object inside column 2.  This doesn't seem like a good thing.  What should be the right thing to do here?  Introduce SqlAccessor.getLocal[Date|Time|Timestamp] accessors to supplement the existing get[Date|Time|Timestamp]?

Reply:
Hmm. That takes us back to the original problem, that of the date|time|timestamp field inside a complex object. 
```
select t.context.date, t.context from test t;

will return a java.sql.Date object for column 1, but a java.time.LocalDate for the same object inside column 2. This doesn't seem like a good thing.
```
Why should that be a bad thing though? Ultimately, the object returned by getObject() is displayed to the end user thru the toString method. The string representation of Local[Date|Time|Timestamp]  should be the same as that of java.sql.[Date|Time|Timestamp]. Isn't it?

Reply:
The string representation between LocalDateTime and Timestamp are not exactly the same, but that is potentially fixable since we can alter the way the values are displayed via formatters.  Though JDBC is not just for getting string representations.  It is on the programmatic use cases where we are getting the value objects where one would see the disconnect on the data types.  Will try this out with a use case I have with programmatic JDBC access and see what are the impacts on different types for the same expected value.